pkg/services/authz/rbac/service.go

```diff
@@ -619,36 +619,65 @@ func (s *Service) listPermission(ctx context.Context, scopeMap map[string]bool,
 		}
 	}
 
-	folderSet := make(map[string]struct{}, len(scopeMap))
+	var res *authzv1.ListResponse
+	if strings.HasPrefix(req.Action, "folders:") {
+		res = buildFolderList(scopeMap, folderMap)
+	} else {
+		res = buildItemList(scopeMap, folderMap, t.prefix())
+	}
```

Comment:
I think we can just move folders to items and remove `buildFolderList`.
```suggestion
	res = buildItemList(scopeMap, folderMap, t.prefix())
	if strings.HasPrefix(req.Action, "folders:") {
		res.Items, res.Folders = res.Folders, []string{}
	}
```